addons/stock/models/stock_move_line.py

```diff
@@ -329,6 +329,7 @@ def write(self, vals):
             mls = self.filtered(lambda ml: ml.move_id.state == 'done' and ml.product_id.type == 'product')
             if not updates:  # we can skip those where qty_done is already good up to UoM rounding
                 mls = mls.filtered(lambda ml: not float_is_zero(ml.qty_done - vals['qty_done'], precision_rounding=ml.product_uom_id.rounding))
+            mls = mls.with_context(to_unlink_candidates=set())
             for ml in mls:
```

Comment:
When I used following code
```suggestion
            to_unlink_candidates=set()
            for ml in mls.with_context(to_unlink_candidates=to_unlink_candidates):
```

`to_unlink_candidates` was never updated

to reproduce:
`-i stock  --test-tags=:StockMove.test_edit_done_move_line_7`